Build a regression model.

In [12]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
import matplotlib.pyplot as plt
import seaborn as sb
df1 = pd.read_csv("pubs_info.csv")
df2 = pd.read_csv("station_details.csv")
df3 = pd.read_csv("yelp_results.csv")
merged_df= pd.merge(df1, df2, on='Station')
merged_df = pd.merge(merged_df, df3, on="Station")

In [24]:
import statsmodels.api as sm
X = sm.add_constant(X) # adding a constant
lin_reg = sm.OLS(y,X['Free Bikes_y'])
model = lin_reg.fit()
print_model = model.summary()

Provide model output and an interpretation of the results. 

In [23]:
print(print_model)

                                 OLS Regression Results                                
Dep. Variable:           Review Count   R-squared (uncentered):                   0.196
Model:                            OLS   Adj. R-squared (uncentered):              0.185
Method:                 Least Squares   F-statistic:                              16.85
Date:                Tue, 19 Dec 2023   Prob (F-statistic):                    0.000109
Time:                        20:00:45   Log-Likelihood:                         -538.96
No. Observations:                  70   AIC:                                      1080.
Df Residuals:                      69   BIC:                                      1082.
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                   coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------

Looking into the correlation of Free bikes and Review Count the following is apparent. 
1. The R-Squared indicates that 19.6% of the variance in Review Count is explain by Free Bikes.
2. The Coefficient is 17.8121 and the low p-value associated with it indicates a possibility that this is statistically significant. This coefficient suggests that each additional unit in Free Bikes is associated with an increase of approximately 17.81 units in Review Count. 

# Stretch

How can you turn the regression model into a classification model?